File: artifacts/zopfli/manual_translation/src/zopfli_block_state.rs
use crate::*;

#[derive(Clone)]
pub struct ZopfliBlockState {
    pub options: Option<Box<ZopfliOptions>>,
    pub lmc: Option<Box<ZopfliLongestMatchCache>>,
    pub blockstart: usize,
    pub blockend: usize,
}

File: artifacts/zopfli/decompose/src/zopfli_block_state.c
typedef struct ZopfliBlockState
{
  const ZopfliOptions *options;
  ZopfliLongestMatchCache *lmc;
  size_t blockstart;
  size_t blockend;
} ZopfliBlockState
/*** DEPENDENCIES:
typedef struct ZopfliLongestMatchCache
{
  unsigned short *length;
  unsigned short *dist;
  unsigned char *sublen;
} ZopfliLongestMatchCache
----------------------------
typedef struct ZopfliOptions
{
  int verbose;
  int verbose_more;
  int numiterations;
  int blocksplitting;
  int blocksplittinglast;
  int blocksplittingmax;
} ZopfliOptions
----------------------------
***/


